libsplinter/src/admin/service/mod.rs

```diff
@@ -523,6 +515,24 @@ impl Service for AdminService {
     }
 }
 
+fn handle_peer_manager_notification(
+    notification: PeerManagerNotification,
+    mut admin_shared: MutexGuard<AdminServiceShared>,
```

Comment:
You should be able to just have this be `&mut AdminServiceShared`, and not need the mutex guard.  Might need to pass in `&mut *admin_shared` at the call site.